code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -223,5 +196,128 @@
+		"tritium" = 10,
+		"plasma" = 10
+	)
+
+
+/datum/gas_reaction/bzformation/react(datum/gas_mixture/air)
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	var/pressure = air.return_pressure()
+
+	var/old_heat_capacity = air.heat_capacity()
+	var/reaction_efficency = pressure/(0.1*ONE_ATMOSPHERE)
+	var/energy_released = 2*reaction_efficency*FIRE_CARBON_ENERGY_RELEASED
+
+
+	ASSERT_GAS("BZ",air)
+	cached_gases["bz"][MOLES]+= reaction_efficency
+	cached_gases["tritium"][MOLES]-= 2*reaction_efficency
+	cached_gases["plasma"][MOLES]-= reaction_efficency
+
+
+	if(energy_released > 0)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.temperature = max(((temperature*old_heat_capacity + energy_released)/new_heat_capacity),TCMB)
+		return REACTING
+
+/datum/gas_reaction/stimformation //Stimulum formation follows a strange pattern of how effective it will be at a given temperature, having some multiple peaks and some large dropoffs. Exo and endo thermic.
+	priority = 3
+	name = "Stimulum formation"
+	id = "stimformation"
+/datum/gas_reaction/stimformation/init_reqs()
+	min_requirements = list(
+		"tritium" = 30,
+		"plasma" = 10,
+		"bz" = 20,
+		"browns" = 30,
+		"temp" = STIMULUM_HEAT_SCALE/2)
+
+/datum/gas_reaction/stimformation/react(datum/gas_mixture/air)
+	var/list/cached_gases = air.gases
+
+	var/old_heat_capacity = air.heat_capacity()
+	var/heat_scale = air.temperature/STIMULUM_HEAT_SCALE
+	var/stim_energy_change
+	stim_energy_change = (STIMULUM_FIRST_RISE(heat_scale**2)) - (STIMULUM_FIRST_DROP(heat_scale**3)) + (STIMULUM_SECOND_RISE(heat_scale**4)) - (STIMULUM_ABSOLUTE_DROP(heat_scale**5))
+
+	ASSERT_GAS("stim",air)
+	cached_gases["stim"][MOLES]+= heat_scale/10
+	cached_gases["tritium"][MOLES]-= heat_scale
+	cached_gases["plasma"][MOLES]-= heat_scale
+	cached_gases["browns"][MOLES]-= heat_scale
+
+	if(stim_energy_change)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.temperature = max(((air.temperature*old_heat_capacity + stim_energy_change)/new_heat_capacity),TCMB)
+		return REACTING
+
+/datum/gas_reaction/nobeliumformation //Hyper-Nobelium formation is extrememly endothermic, but requires high temperatures to start. Due to its high mass, hyper-nobelium uses large amounts of nitrogen and tritium. BZ can be used as a catalyst to make it less endothermic.
+	priority = 3
+	name = "Hyper-Nobelium condensation"
+	id = "nobformation"
+
+/datum/gas_reaction/nobeliumformation/init_reqs()
+	min_requirements = list(
+		"nitrogen" = 10,
+		"tritium" = 5,
+		"temp" = 500000)
+
+/datum/gas_reaction/nobeliumformation/react(datum/gas_mixture/air)
+	var/list/cached_gases = air.gases
+	air.assert_gases("nob","bz")
+	var/old_heat_capacity = air.heat_capacity()
+	var/nob_formed = (cached_gases["nitrogen"][MOLES]*cached_gases["tritium"][MOLES])/100
+	var/energy_taken = nob_formed*(1000000/cached_gases["bz"][MOLES])
```

Comment:
Uh, won't bz moles be zero if there was none before.

Reply:
The idea is the reaction doesn't require BZ to happen, but if it is there it will make it less endothermic. But if you mean that there's a potential to divide by zero, yeah I just noticed that. I'll change that around for safety reasons.